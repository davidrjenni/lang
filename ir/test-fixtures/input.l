{
	assert ~(~(true));
	assert ~(~(false));
	assert 27 = 3 + 5*5 - 1;
	assert false = true | true;
	assert -1 = 0 - 1;

	for true {
		assert -1 = 0 - 1;
	}

	if true {
		assert true;
	}

	for true {
		if true {
			break;
		}
	}

	for true {
		if true {
			for true {
				break;
				assert true;
			}
			continue;
		}
	}
}
