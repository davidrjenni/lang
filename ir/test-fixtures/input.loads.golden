load rbool.0 <- bool(true)  // test-fixtures/input.l:2:12
cmp rbool.0 bool(true)  // test-fixtures/input.l:2:12
setne rbool.0  // test-fixtures/input.l:2:11
cmp rbool.0 bool(true)  // test-fixtures/input.l:2:10
setne rbool.0  // test-fixtures/input.l:2:9
cmp rbool.0 bool(true)  // test-fixtures/input.l:2:9
cjump .L1  // test-fixtures/input.l:2:2
load ri64.1 <- i64(2)  // test-fixtures/input.l:2:2
call AssertViolated  // test-fixtures/input.l:2:2
.L1
load rbool.0 <- bool(false)  // test-fixtures/input.l:3:12
cmp rbool.0 bool(true)  // test-fixtures/input.l:3:12
setne rbool.0  // test-fixtures/input.l:3:11
cmp rbool.0 bool(true)  // test-fixtures/input.l:3:10
setne rbool.0  // test-fixtures/input.l:3:9
cmp rbool.0 bool(true)  // test-fixtures/input.l:3:9
cjump .L2  // test-fixtures/input.l:3:2
load ri64.1 <- i64(3)  // test-fixtures/input.l:3:2
call AssertViolated  // test-fixtures/input.l:3:2
.L2
load ri64.0 <- i64(5)  // test-fixtures/input.l:4:18
load ri64.1 <- i64(5)  // test-fixtures/input.l:4:18
mul ri64.0 ri64.1  // test-fixtures/input.l:4:18
push ri64.0  // test-fixtures/input.l:4:14
load ri64.0 <- i64(3)  // test-fixtures/input.l:4:14
pop ri64.1  // test-fixtures/input.l:4:14
add ri64.0 ri64.1  // test-fixtures/input.l:4:14
load ri64.1 <- i64(1)  // test-fixtures/input.l:4:14
sub ri64.0 ri64.1  // test-fixtures/input.l:4:14
push ri64.0  // test-fixtures/input.l:4:9
load ri64.0 <- i64(27)  // test-fixtures/input.l:4:9
pop ri64.1  // test-fixtures/input.l:4:9
cmp ri64.0 ri64.1  // test-fixtures/input.l:4:9
sete rbool.0  // test-fixtures/input.l:4:9
cmp rbool.0 bool(true)  // test-fixtures/input.l:4:9
cjump .L3  // test-fixtures/input.l:4:2
load ri64.1 <- i64(4)  // test-fixtures/input.l:4:2
call AssertViolated  // test-fixtures/input.l:4:2
.L3
load rbool.0 <- bool(false)  // test-fixtures/input.l:5:9
load rbool.1 <- bool(true)  // test-fixtures/input.l:5:9
cmp rbool.0 rbool.1  // test-fixtures/input.l:5:9
sete rbool.0  // test-fixtures/input.l:5:9
load rbool.1 <- bool(true)  // test-fixtures/input.l:5:9
or rbool.0 rbool.1  // test-fixtures/input.l:5:9
cmp rbool.0 bool(true)  // test-fixtures/input.l:5:9
cjump .L4  // test-fixtures/input.l:5:2
load ri64.1 <- i64(5)  // test-fixtures/input.l:5:2
call AssertViolated  // test-fixtures/input.l:5:2
.L4
load ri64.0 <- i64(0)  // test-fixtures/input.l:6:14
load ri64.1 <- i64(1)  // test-fixtures/input.l:6:14
sub ri64.0 ri64.1  // test-fixtures/input.l:6:14
push ri64.0  // test-fixtures/input.l:6:9
load ri64.0 <- i64(1)  // test-fixtures/input.l:6:9
neg ri64.0  // test-fixtures/input.l:6:9
pop ri64.1  // test-fixtures/input.l:6:9
cmp ri64.0 ri64.1  // test-fixtures/input.l:6:9
sete rbool.0  // test-fixtures/input.l:6:9
cmp rbool.0 bool(true)  // test-fixtures/input.l:6:9
cjump .L5  // test-fixtures/input.l:6:2
load ri64.1 <- i64(6)  // test-fixtures/input.l:6:2
call AssertViolated  // test-fixtures/input.l:6:2
.L5
.L6
load rbool.0 <- bool(true)  // test-fixtures/input.l:8:6
cmp rbool.0 bool(false)  // test-fixtures/input.l:8:6
cjump .L7  // test-fixtures/input.l:8:2
load ri64.0 <- i64(0)  // test-fixtures/input.l:9:15
load ri64.1 <- i64(1)  // test-fixtures/input.l:9:15
sub ri64.0 ri64.1  // test-fixtures/input.l:9:15
push ri64.0  // test-fixtures/input.l:9:10
load ri64.0 <- i64(1)  // test-fixtures/input.l:9:10
neg ri64.0  // test-fixtures/input.l:9:10
pop ri64.1  // test-fixtures/input.l:9:10
cmp ri64.0 ri64.1  // test-fixtures/input.l:9:10
sete rbool.0  // test-fixtures/input.l:9:10
cmp rbool.0 bool(true)  // test-fixtures/input.l:9:10
cjump .L8  // test-fixtures/input.l:9:3
load ri64.1 <- i64(9)  // test-fixtures/input.l:9:3
call AssertViolated  // test-fixtures/input.l:9:3
.L8
jump .L6  // test-fixtures/input.l:8:2
.L7
load rbool.0 <- bool(true)  // test-fixtures/input.l:12:5
cmp rbool.0 bool(false)  // test-fixtures/input.l:12:5
cjump .L9  // test-fixtures/input.l:12:2
load rbool.0 <- bool(true)  // test-fixtures/input.l:13:10
cmp rbool.0 bool(true)  // test-fixtures/input.l:13:10
cjump .L10  // test-fixtures/input.l:13:3
load ri64.1 <- i64(13)  // test-fixtures/input.l:13:3
call AssertViolated  // test-fixtures/input.l:13:3
.L10
.L9
.L11
load rbool.0 <- bool(true)  // test-fixtures/input.l:16:6
cmp rbool.0 bool(false)  // test-fixtures/input.l:16:6
cjump .L12  // test-fixtures/input.l:16:2
load rbool.0 <- bool(true)  // test-fixtures/input.l:17:6
cmp rbool.0 bool(false)  // test-fixtures/input.l:17:6
cjump .L13  // test-fixtures/input.l:17:3
jump .L12  // test-fixtures/input.l:18:4
.L13
jump .L11  // test-fixtures/input.l:16:2
.L12
.L14
load rbool.0 <- bool(true)  // test-fixtures/input.l:22:6
cmp rbool.0 bool(false)  // test-fixtures/input.l:22:6
cjump .L15  // test-fixtures/input.l:22:2
load rbool.0 <- bool(true)  // test-fixtures/input.l:23:6
cmp rbool.0 bool(false)  // test-fixtures/input.l:23:6
cjump .L16  // test-fixtures/input.l:23:3
.L17
load rbool.0 <- bool(true)  // test-fixtures/input.l:24:8
cmp rbool.0 bool(false)  // test-fixtures/input.l:24:8
cjump .L18  // test-fixtures/input.l:24:4
jump .L18  // test-fixtures/input.l:25:5
load rbool.0 <- bool(true)  // test-fixtures/input.l:26:12
cmp rbool.0 bool(true)  // test-fixtures/input.l:26:12
cjump .L19  // test-fixtures/input.l:26:5
load ri64.1 <- i64(26)  // test-fixtures/input.l:26:5
call AssertViolated  // test-fixtures/input.l:26:5
.L19
jump .L17  // test-fixtures/input.l:24:4
.L18
jump .L14  // test-fixtures/input.l:28:4
.L16
jump .L14  // test-fixtures/input.l:22:2
.L15
